From 7b807843e011a003e4e61c61ea17e0d9184dad20 Mon Sep 17 00:00:00 2001
From: Laurent Bigonville <bigon@bigon.be>
Date: Thu, 25 Mar 2010 13:22:29 +0100
Subject: [PATCH] Fix symbols visibility

---
 ConfigureChecks.cmake |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/ConfigureChecks.cmake b/ConfigureChecks.cmake
index c2bed2e..07fb7bf 100644
--- a/ConfigureChecks.cmake
+++ b/ConfigureChecks.cmake
@@ -16,9 +16,9 @@ set(SYSCONFDIR ${SYSCONF_INSTALL_DIR})
 set(BINARYDIR ${CMAKE_BINARY_DIR})
 set(SOURCEDIR ${CMAKE_SOURCE_DIR})
 
-if(CMAKE_COMPILER_IS_GNUC)
+if(CMAKE_COMPILER_IS_GNUCC)
 check_c_compiler_flag("-fvisibility=hidden" WITH_VISIBILITY_HIDDEN)
-endif(CMAKE_COMPILER_IS_GNUC)
+endif(CMAKE_COMPILER_IS_GNUCC)
 
 # HEADER FILES
 check_include_file(argp.h HAVE_ARGP_H)
-- 
1.7.0

