Description: Be sure we never build LaTeX documentation
Author: Laurent Bigonville <bigon@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=622108

--- a/doc/doxy.config.in
+++ b/doc/doxy.config.in
@@ -1014,7 +1014,7 @@
 # If the GENERATE_LATEX tag is set to YES (the default) Doxygen will
 # generate Latex output.
 
-GENERATE_LATEX         = @DOXYFILE_LATEX@
+GENERATE_LATEX         = NO
 
 # The LATEX_OUTPUT tag is used to specify where the LaTeX docs will be put.
 # If a relative path is entered the value of OUTPUT_DIRECTORY will be
