From de706de8c3f67ae8408b2c8be139d177ef5d567c Mon Sep 17 00:00:00 2001
From: milo <milo@r0ot.me>
Date: Mon, 14 Feb 2011 19:23:12 +0000
Subject: Check for NULL pointers in string.c

(cherry picked from commit 4230509e805f546091ca2e4f8aa13c22f6efc7a8)
---
--- a/src/string.c
+++ b/src/string.c
@@ -21,6 +21,7 @@
  * MA 02111-1307, USA.
  */
 
+#include <errno.h>
 #include <stdlib.h>
 #include <string.h>
 
@@ -91,8 +92,15 @@
  * @note The nul byte is not copied nor counted in the ouput string.
  */
 struct ssh_string_struct *ssh_string_from_char(const char *what) {
-  struct ssh_string_struct *ptr = NULL;
-  size_t len = strlen(what);
+  struct ssh_string_struct *ptr;
+  size_t len;
+
+  if(what == NULL) {
+      errno = EINVAL;
+      return NULL;
+  }
+
+  len = strlen(what);
 
   ptr = malloc(4 + len);
   if (ptr == NULL) {
@@ -133,7 +141,7 @@
 char *ssh_string_to_char(struct ssh_string_struct *s) {
 	size_t len;
 	char *new;
-	if(s==NULL)
+	if(s==NULL || s->string == NULL)
 		return NULL;
   len = ntohl(s->size) + 1;
   new = malloc(len);
@@ -164,7 +172,12 @@
  * @return              Newly allocated copy of the string, NULL on error.
  */
 struct ssh_string_struct *ssh_string_copy(struct ssh_string_struct *s) {
-  struct ssh_string_struct *new = malloc(ntohl(s->size) + 4);
+  struct ssh_string_struct *new;
+  
+  if(s == NULL || s->string == NULL) {
+      return NULL;
+  }
+  new = malloc(ntohl(s->size) + 4);
 
   if (new == NULL) {
     return NULL;
